File: passenger-service/src/main/java/io/voitovich/yura/passengerservice/event/service/KafkaChannelGateway.java
package io.voitovich.yura.passengerservice.event.service;


import io.voitovich.yura.passengerservice.event.model.ConfirmRatingReceiveModel;
import org.springframework.integration.annotation.Gateway;
import org.springframework.integration.annotation.MessagingGateway;

@MessagingGateway
public interface KafkaChannelGateway {

    @Gateway(requestChannel = "confirmRatingReceiveChannel")
    void handleRatingReceiveMessage(ConfirmRatingReceiveModel model);
}

File: ride-service/src/main/kotlin/io/voitovich/yura/rideservice/event/service/KafkaChannelGateway.kt
package io.voitovich.yura.rideservice.event.service

import io.voitovich.yura.rideservice.event.model.SendRatingModel
import org.springframework.integration.annotation.Gateway
import org.springframework.integration.annotation.MessagingGateway


@MessagingGateway
interface KafkaChannelGateway : SendRatingHandler {

    @Gateway(requestChannel = "ratePassengerChannel")
    override fun handleRatePassengerRequest(sendRatingModel: SendRatingModel)


    @Gateway(requestChannel = "rateDriverChannel")
    override fun handleRateDriverRequest(sendRatingModel: SendRatingModel)
}